File: src/main/java/com/github/windchopper/tools/log/browser/Launcher.java
package com.github.windchopper.tools.log.browser;

import com.github.windchopper.common.fx.cdi.ResourceBundleLoad;
import com.github.windchopper.common.fx.cdi.form.FormLoader;
import com.github.windchopper.common.fx.cdi.form.StageFormLoad;
import com.github.windchopper.common.util.ClassPathResource;
import javafx.application.Application;
import javafx.stage.Stage;
import org.jboss.weld.environment.se.Weld;
import org.jboss.weld.environment.se.WeldContainer;

public class Launcher extends Application {

    private Weld weld;
    private WeldContainer weldContainer;

    @Override public void init() throws Exception {
        super.init();
        weld = new Weld()
            .enableDiscovery()
            .addPackages(FormLoader.class);
        weldContainer = weld.initialize();
    }

    @Override public void stop() throws Exception {
        weld.shutdown();
        super.stop();
    }

    @Override public void start(Stage primaryStage) {
        var beanManager = weldContainer.getBeanManager();

        beanManager.fireEvent(
            new ResourceBundleLoad(Globals.bundle));
        beanManager.fireEvent(
            new StageFormLoad(
                new ClassPathResource(Globals.FXML__MAIN),
                () -> primaryStage));
    }

    /*
     *
     */

    public static void main(String... args) {
        launch(args);
    }

}

File: src/main/java/com/github/windchopper/tools/log/browser/Launcher.kt
package com.github.windchopper.tools.log.browser

import com.github.windchopper.common.fx.cdi.ResourceBundleLoad
import com.github.windchopper.common.fx.cdi.form.FormLoader
import com.github.windchopper.common.fx.cdi.form.StageFormLoad
import com.github.windchopper.common.util.ClassPathResource
import javafx.application.Application
import javafx.stage.Stage
import org.jboss.weld.environment.se.Weld
import org.jboss.weld.environment.se.WeldContainer

class Launcher: Application() {

    private var weld: Weld = Weld()
        .enableDiscovery()
        .addPackages(FormLoader::class.java)
    private var weldContainer: WeldContainer = weld.initialize()

    override fun stop() {
        weld.shutdown()
        super.stop()
    }

    override fun start(primaryStage: Stage) {
        with (weldContainer.beanManager) {
            fireEvent(ResourceBundleLoad(Globals.bundle))
            fireEvent(StageFormLoad(ClassPathResource(Globals.FXML__MAIN)) { primaryStage })
        }
    }

}

fun main(vararg args: String) {
    Application.launch(*args)
}
